Content.Server/Projectiles/ReflectProjectileComponent.cs

```diff
@@ -0,0 +1,23 @@
+using Robust.Shared.Audio;
+
+namespace Content.Server.Projectiles
+{
+    /// <summary>
+    ///     Entities with this component have a chance to reflect projectiles
+    /// </summary>
+    [RegisterComponent]
+    public sealed class ReflectProjectileComponent : Component 
+    {
+        /// <summary>
+        ///     Can only reflect when enabled
+        /// </summary>
+        [DataField("enabled")]
+        public bool Enabled;
+
+        [DataField("reflectChance")]
+        public float ReflectChance;
+
+        [DataField("onReflect")]
+        public SoundSpecifier? OnReflect { get; set; }
+    }
+}
```

Comment:
This should be using file-scoped namespaces, in shared and networked, also how come enabled is false by default and one of the datafields is a property?